apis/projectcontour/v1alpha1/contourdeployment.go

```diff
@@ -123,6 +123,14 @@ type EnvoySettings struct {
 	//
 	// +optional
 	NodePlacement *NodePlacement `json:"nodePlacement,omitempty"`
+
+	// ExtraVolumes holds the extra volumes to add.
+	// +optional
+	ExtraVolumes []corev1.Volume `json:"extraVolumes,omitempty"`
+
+	// ExtraVolumeMounts holds the extra volume mounts to add (normally used with extraVolumes).
+	// +optional
+	ExtraVolumeMounts []corev1.VolumeMount `json:"extraVolumeMounts,omitempty"`
```

Comment:
I think we should add validation logic to the GatewayClass controller within the provisioner, to ensure that any ExtraVolumeMounts reference a volume from ExtraVolumes -- does that make sense? (In here: https://github.com/projectcontour/contour/blob/main/internal/provisioner/controller/gatewayclass.go#L159)

Reply:
yep good idea +1

Reply:
i will upgrade it later